### ISSUE TITLE
Python code with pyroscope failed

### ISSUE THREAD

**Issue by jobber2955** [pyroscope-io/pyroscope#809]
Run any python code(even if the example one) will come with a pyroscope failure:```failed to find python version from target process pyroscope```

And also tried pycharm, same problem

This problem does not happen in Go.

Any help?

**Comment by Rperry2174** [comment]
Hi @jobber2955 thanks for reporting and sorry it's not working! Can you give a little more information?

- What system are you running it on? 
- Which version of python is installed? `which python` 
- Have you tried running it in docker-compose? Examples like [this one](https://github.com/pyroscope-io/pyroscope/tree/main/examples/python/flask) should work fine 
- Are your running `pyroscope exec python script.py` or are you using the pip package?

**Comment by jobber2955** [comment]
@Rperry2174 Thanks for replying. Here's more information


System: MacOS 12
Python Version: 3.8

Sample code:
```
if __name__ == '__main__':
    web=FastAPI()
    init_jvm()
    init_route()

    # If you need profiling, use pyroscope below
    pyroscope.configure(
        app_name="android_jar_api",  # replace this with some name for your application
        server_address="http://127.0.0.1:4040",  # replace this with the address of your pyroscope server
    )

    # In debug, uncommon below lines to debug in pycharm.
    uvicorn.run(web, host="0.0.0.0", port=8080)
```

Im using pip package, and haven't tried the docker one

**Comment by jobber2955** [comment]
Using ```pyroscope exec python XXX.py``` i got more information:

```
INFO[2022-02-09T14:35:49.539109] exec/cli.go:50 to disable logging from pyroscope, specify -no-logging flag 
INFO[2022-02-09T14:35:49.539273] exec/cli.go:57 we recommend specifying application name via -application-name flag or env variable PYROSCOPE_APPLICATION_NAME 
INFO[2022-02-09T14:35:49.539329] exec/cli.go:60 for now we chose the name for you and it's "pyspy.agitated.newton" 
INFO[2022-02-09T14:35:49.539875] exec/cli_unix.go:51 dropping permissions, running command as "jobber" (501/20) 
Error: start session: Failed to open process - check if it is running.
[ERROR] error happened when starting profiling session: Failed to find python version from target process
```
My script is a simple FastAPI server, using JPype1 to get some result and return to client.

**Comment by abeaumont** [comment]
> Im using pip package, and haven't tried the docker one
> Using `pyroscope exec python XXX.py` i got more information:

When using the pip package there's no need to use `pyroscope exec` (I suspect that may be causing the error), as the pip package takes care of sending the profiling data, running `python XXX.py` should be enough. Let us know if that fixes the problem.

**Comment by jobber2955** [comment]
@abeaumont 
Using ```python3 xxx.py``` shows 
```[ERROR] error happened when starting profiling session: Failed to find a python interpreter in the .data section```

And using pycharm shows
```[ERROR] error happened when starting profiling session: Failed to find python version from target process```

Will this be a path encoding problem? My path-to-python contains Chinese character, I will try change the path to all English and feedback

**Comment by abeaumont** [comment]
@jobber2955 this error means that py-spy is unable to detect the python version from both the CLI and the process memory. It should be unrelated to the encoding, but hard to tell the reason. As a workaround, it should work if you use the fully versioned python binary: `python3.8 xxx.py`

**Comment by jobber2955** [comment]
@abeaumont This workaround works, thanks. But here's another question: In python pyroscope will only profile cpu? In official docs there's no configuration like Go's ```pyroscope.ProfileAllocObjects```

**Comment by abeaumont** [comment]
@jobber2955 yeah, that's correct. Currently, only CPU profiling is supported in Python. For memory profiling, there's an open issue that can be used to track its progress: https://github.com/pyroscope-io/pyroscope/issues/316
